File: framework/languages/java/e2e/SpotService/src/main/java/systems/zlink/e2e/spotservice/IngressCommandHandler.java
package systems.zlink.e2e.spotservice;

import systems.zlink.framework.channels.ZLinkSendContext;
import systems.zlink.framework.channels.ZLinkSendHandler;

public final class IngressCommandHandler implements ZLinkSendHandler<Contracts.OutboundCommand> {
    private final ScenarioState state;

    public IngressCommandHandler(ScenarioState state) {
        this.state = state;
    }

    @Override
    public void handle(
        Contracts.OutboundCommand message,
        ZLinkSendContext context) {
        state.record("IngressCommand", "channel", message.value());
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/IngressCommandHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.channels.ZLinkSendContext
import systems.zlink.framework.kotlin.ZLinkSuspendingSendHandler

class IngressCommandHandler(
    private val state: ScenarioState,
) : ZLinkSuspendingSendHandler<Contracts.OutboundMsg> {
    override suspend fun handle(message: Contracts.OutboundMsg, context: ZLinkSendContext) {
        state.record("IngressCommand", "channel", message.value)
    }
}
